src/main/java/nextstep/fp/Lambda.java

```diff
@@ -27,30 +27,19 @@ public void run() {
     }
 
     public static int sumAll(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            total += number;
-        }
-        return total;
+        return sumConditional(numbers, (number) -> true);
     }
 
     public static int sumAllEven(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number % 2 == 0) {
-                total += number;
-            }
-        }
-        return total;
+        return sumConditional(numbers, (number) -> number % 2 == 0);
     }
 
     public static int sumAllOverThree(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number > 3) {
-                total += number;
-            }
-        }
-        return total;
+        return sumConditional(numbers, (number) -> number > 3);
+
+    }
+
+    private static int sumConditional(List<Integer> numbers, Conditional conditional) {
+        return numbers.stream().filter(conditional::test).reduce(0, (x, y) -> x + y);
```

Comment:
reduce 사용 👍

stream 사용시엔 dot 하나당 개행을 하여 가독성을 높혀보시는것도 좋을것 같습니다.

```suggestion
        return numbers.stream()
            .filter(conditional::test)
            .reduce(0, Integer::sum);
```